extrabees/src/main/java/binnie/extrabees/worldgen/WorldGenHiveMarble.java

```diff
@@ -47,4 +49,47 @@ public boolean generate(World world, Random rand, BlockPos position) {
 
 		return true;
 	}
+
+	@Nullable
+	@Override
+	public BlockPos getPosForHive(World world, int x, int z) {
+		//get to the ground
+		final BlockPos topPos = world.getHeight(new BlockPos(x, 0, z));
+		if (topPos.getY() == 0) {
+			return null;
+		}
+
+		final BlockPos.MutableBlockPos pos = new BlockPos.MutableBlockPos(topPos);
+
+		while (pos.getY() > 0) {
+			if (isValidLocation(world, pos)) {
+				return pos;
+			}
+			pos.add(0, -1, 0);
```

Comment:
The strategy here will check every block from the surface down to 0.
A more efficient strategy would be to get the  `topPos` and try a few random locations between 0 and `topPos`.
Only try a few locations before giving up.